chroot-bin/volume

```diff
@@ -0,0 +1,179 @@
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+set -e
+set -u
+
+DEFAULT_VOLUME_DELTA=5
+
+get_current_output() {
+    # This function gets the current Output Node ID (x:y)
+
+    # From cras_test_client we can get the current output node from
+    # a specific line. This sed pipeline extracts it.
+    # The line looks like this:
+    #     Selected Output Node: 7:1
+    cras_test_client --dump_server_info | sed -n \
+        -e '/Selected Output Node/s#[[:space:]]*##g;s#^[a-zA-Z]*:##p'
+    #       ^                     ^                 ^              ^ print
+    #       ^                     ^                 ^ substitute letters
+    #       ^                     ^ substitude spaces for nothing
+    #       ^ select this line only
+}
+
+get_current_output_infos() {
+    # This function gets line from cras_test_client --dump_server_info that 
+    # represents the current state of the current output.
+
+    # The current output informations will not duplicate an output node, this makes
+    # it safe(-ish) to filter out on the current output ID.
+    # This would also select the "Selected Output Node" line, but by filtering the
+    # lines beginning with space characters, this selects the proper line.
+    # If for any reason this starts failing, it could be replaced with a sed script
+    # that acts only between the two "headers" "Output Nodes" and "Input Devices".
+    cras_test_client --dump_server_info | grep "^[[:space:]].*$(get_current_output)"
+}
+
+get_current_volume() {
+    get_current_output_infos | sed -e \
+        "s#[[:space:]]\+#;#g" | cut -d';' -f3
+    #    ^                        ^ Use cut to get the third field (Volume)
+    #    ^ normalize spaces/tab separators to semicolons
+}
+
+set_volume() {
+    local volume="$1"
+    # Clamp volume to zero
+    # A negative amount of absolute volume sets it to 100%
+    if [ "$volume" -lt 0 ] ; then
+        volume=0
+    fi
+    cras_test_client --set_node_volume "$(get_current_output):$volume"
+}
+
+get_is_muted() {
+    local counter=0
+    local muted="$(host-dbus dbus-send --print-reply --system \
+        --dest=org.chromium.cras --type=method_call \
+        /org/chromium/cras org.chromium.cras.Control.GetVolumeState \
+        | sed -n -e '6s/.*boolean[[:space:]]\+//p')"
+    [ "$muted" = "true" ]
```

Comment:
It's a little nicer to treat functions like functions, and explicitly return 1 or 0 depending on the outcome of that test.